resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -215,8 +232,10 @@ private[spark] class Client(
     sparkConf.setIfMissing("spark.driver.port", DEFAULT_DRIVER_PORT.toString)
     sparkConf.setIfMissing("spark.blockmanager.port",
       DEFAULT_BLOCKMANAGER_PORT.toString)
+    sparkConf.set("spark.ui.basePath", s"/$kubernetesAppId/$UI_PATH_COMPONENT")
```

Comment:
The more I think about it the more I think the base path should be consistent between Ingress mode and non-Ingress mode. It would be nice to avoid a base path entirely, but unfortunately the Ingress will take the path "as-is" and forward that exact path to the backend by default. Some ingress controllers support path rewriting, but not all Ingress controllers would indeed support that so we have to go with the lowest common denominator, unfortunately.